test/tls_test.go

```diff
@@ -1819,6 +1819,40 @@ func TestTLSPinnedCertsClient(t *testing.T) {
 	nc.Close()
 }
 
+func TestTLSConnectionRate(t *testing.T) {
+	config := `
+		listen: "127.0.0.1:-1"
+		tls {
+			cert_file: "./configs/certs/server-cert.pem"
+			key_file:  "./configs/certs/server-key.pem"
+			connection_rate_limit: 3
+		}
+	`
+
+	confFileName := createConfFile(t, []byte(config))
+	defer removeFile(t, confFileName)
+
+	srv, _ := RunServerWithConfig(confFileName)
+	defer srv.Shutdown()
+
```

Comment:
Here you could add a debug logger that captures warnings, which I then think that would have shown that the logger would not work (because of startGoRoutine() invoked in NewServer()).

Something like this:
```
l := &captureWarnLogger{warn: make(chan string, 100)}
s.SetLogger(l, false, false)
```

Then at the end of this test, check that you got some of the warnings you expect:
```
for {
	select {
	case w := <-l.warn:
		if strings.Contains(w, "connections due to TLS rate limiting") {
			return
		}
	case <-time.After(time.Second):
		t.Fatal("Did not get warning")
	}
}
```